webui_runner/plugs/wamr_app_aot/shootout/wrapper.c

```diff
@@ -0,0 +1,70 @@
+#include <sightglass.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <time.h>
+#include <unistd.h>
+
+#define STR1(x) #x
+#define STR(x) STR1(x)
+
+#ifdef WORKLOAD_LOCATION
+#define WASM_FILE_PREFIX STR(WORKLOAD_LOCATION)
+#else
+#error "WORKLOAD_LOCATION not defined"
+#endif
+
+#ifdef VM_LOCATION
+#define VM STR(VM_LOCATION)
+#else
+#error "VM_LOCATION not defined"
+#endif
+
+#define MAX_WASM_FILE_NAME_SIZE 500
+
+extern TestsConfig tests_config;
+
+TestsConfig tests_config = {.global_setup = NULL,
+                            .global_teardown = NULL,
+                            .version = TEST_ABI_VERSION};
+
+static void body_wrapper(const char *name, void *ctx)
+{
+    (void)ctx;
+    char wasm_file[MAX_WASM_FILE_NAME_SIZE];
+    snprintf(wasm_file, sizeof(wasm_file), "%s %s/%s", VM, WASM_FILE_PREFIX, name);
```

Comment:
Don't you want this behind a flag or something to avoid polluting the results with IO?

Reply:
Oh.... my mistake: I didn't read closely enough: I thought this was actually printing something. But I guess the same concern applies: we could be concatenating these strings in the `BODY` macro so the concatenation is static instead of dynamic?

Reply:
Not sure I understand the suggestion here? What would go in the Body macro and what would be static instead of dynamic? This template, the way this is written, is from the baseline shootout template so it's not specific to this wamr file.

Reply:
Something like this (untested): 
```
#define BODY(NAME) \
    void NAME##_body(void *ctx) { system(#VM_LOCATION " " #WORKLOAD_LOCATION "/" #NAME ".wasm.aot", ctx); }
```

It would reduce the amount of code and completely avoid any slight overhead with `snprintf`. More info [here](https://gcc.gnu.org/onlinedocs/cpp/Concatenation.html).

Reply:
Ahh, yes, maybe. Unless there is something about how these preprocessor directives that won't allow this. Like some of the other comments, this will apply generally so we can test this out in a future patch that applies consistently in all wrappers.